src/main/client/get_cdtctx_base64.c

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright 2022 Aerospike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+#include <Python.h>
+
+#include <aerospike/aerospike_info.h>
+#include <aerospike/as_error.h>
+#include <aerospike/as_node.h>
+#include <aerospike/as_cluster.h>
+
+#include "client.h"
+#include "conversions.h"
+#include "exceptions.h"
+#include "policy.h"
+
+/**
+ *******************************************************************************************************
+ * Return the base64 representation of the passed in CDT ctx.
+ *
+ * @param self                  AerospikeClient object
+ * @param args                  The args is a tuple object containing an argument
+ *                              list passed from Python to a C function
+ * @param kwds                  Dictionary of keywords
+ *
+ * Returns a string containing the base64 representation of the passed in CDT ctx.
+ * In case of error, appropriate exceptions will be raised.
+ *******************************************************************************************************
+ */
+PyObject *AerospikeClient_GetCDTCTXBase64(AerospikeClient *self,
+											  PyObject *args, PyObject *kwds)
+{
+	// function args
+	PyObject *py_cdtctx = NULL;
+	as_cdt_ctx ctx;
+	bool ctx_in_use = false;
+
+	char *base64 = NULL;
+	PyObject *py_response = NULL;
+
+	as_error err;
+	as_error_init(&err);
+
+	static char *kwlist[] = {"ctx", NULL};
+	if (PyArg_ParseTupleAndKeywords(args, kwds, "O:get_cdtctx_base64",
+									kwlist, &py_cdtctx) == false) {
+		return NULL;
+	}
+
+	//convert filter to base64
+	if (py_cdtctx == NULL || !PyDict_Check(py_cdtctx)) {
```

Comment:
The Python _cdt_ctx objects aren't compiled, just returned by helper functions in the aerospike_helpers.cdt_ctx module. 

Looks like this expects a dictionary, not a list, so the error message should change.

Note: Operations take a list of_cdt_ctx, I just noticed that this and the cdt index methods take dictionaries. Lets make sure the argument format for these is clear in the docs.